corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -625,6 +625,15 @@ def render(self, context):
     return AddToBlockNode(nodelist, 'js-inline')
 
 
+@register.simple_tag
+def html_attr(value):
+    if isinstance(value, basestring):
+        value = json.dumps(value)[1:-1]
```

Comment:
You're right, thanks for flagging - the final value will still have an undesired backslash (`abc\"def`). Initial page data uses the same logic but takes care of this issue by doing a [JSON.parse on get](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/static/hqwebapp/js/initial_page_data.js#L36). It's going to be confusing if data attributes set with `html_attr` need to be parsed when accessed, but data attributes set normally don't need to be parsed. Will think about this some more.

Escaping before json encoding unfortunately doesn't work, because the original value may not be a string.